src/charm.py

```diff
@@ -161,13 +173,34 @@ def _layer(self) -> Layer:
                     "override": "replace",
                     "level": "alive",
                     "exec": {
-                        "command": f"echo ruok | nc {self.state.unit_server.host} {CLIENT_PORT}"
+                        "command": f"echo ruok | nc {self.state.unit_server.internal_address} {CLIENT_PORT}"
                     },
                 }
             },
         }
         return Layer(layer_config)
 
+    def update_external_services(self) -> None:
+        """Attempts to update any external Kubernetes services."""
+        if not SUBSTRATE == "k8s" or not self.unit.is_leader():
+            return
+
+        match self.config.expose_external:
+            case ExposeExternal.FALSE:
+                # if is already removed, will silently continue
+                self.k8s_manager.remove_service(service_name=self.k8s_manager.exposer_service_name)
+                return
+
+            case ExposeExternal.NODEPORT:
+                self.k8s_manager.apply_service(service=self.k8s_manager.build_nodeport_service())
+
+            case ExposeExternal.LOADBALANCER:
+                self.k8s_manager.apply_service(
+                    service=self.k8s_manager.build_loadbalancer_service()
+                )
+
+        self.unit.status = MaintenanceStatus("waiting for service")
```

Comment:
I apologize, would you mind clarifying what you meant there?

Reply:
I mean that this status gets overriden by subsequent logic that for instance calls:

```
self.unit.status = ActiveStatus()
```

Then we would override this. Also I just noticed that we are not setting the `app.status` into Maintenance actually

Reply:
I think `apply_service` is a blocking call. We should probably set `MaintenanceStatus(waiting for service)` **before** calling `apply_service`. I gather from other teams that setting a LB service on public clouds can take many minutes, so that status will be 'live' for long enough to not be overwritten by the next event.